spec-draft.md

```diff
@@ -136,6 +136,106 @@ Example payload for setting "Airplane Mode":
 Data is the upper bits since in the future we may want to support setting
 certain types of Data the device is capable of. For example 3G, 4G, LTE.
 
+Device Orientation
+------------------
+
+The [JSON Wire Protocol](https://code.google.com/p/selenium/wiki/JsonWireProtocol#/session/:sessionId/orientation)
+already supports a very basic concept of
+orientation, involving two orientation options: portrait and landscape.
+Ultimately this will be subsumed under Device Rotation (see below), but for now
+we continue to support these endopints for mobile devices.
+
+* GET /session/:sessionid/orientation
+    * returns ScreenOrientation
+* POST /session/:sessionid/orientation
+    * accepts a ScreenOrientation
+
+### ScreenOrientation
+
+A `ScreenOrientation` can be one of two strings:
+
+* `LANDSCAPE`
+* `PORTRAIT`
+
+Device Rotation
+---------------
+
+Mobile devices can theoretically be used in any spatial orientation. Thus the
+concept of 'orientation' is actually much more general. We will call it
+'rotation'. Currently, mobile OS automation technologies do not expose hooks
+for arbitrary rotation, but allow simply for a small set of 'orientations',
+like landscape or portrait (largely because these are the main categories used
+by apps themselves). We described some simple 'orientation' options above. Here
+we describe the spec for rotation, which can be used when automation tool
+vendors release support for more than 'landscape' and 'portrait' orientations.
+Apple, for example, already has more options (such as "portrait + faceup").
+
+We can define any rotation with a combination of 3 angles: rotations around the
+X, Y, and Z axes in 3d space. The X axis is horizontal (width), Y is vertical
+(height), and Z is extensive (depth). We arbitrarily stipulate the "default"
+position for a device (i.e., its position when the X, Y and Z angles are 0, 0,
+and 0) is as follows:
+
+* The device screen is facing us, parallel to the value 0 on the Z axis (i.e.,
+  it extends into the positive Z space, not the negative (see below))
```

Comment:
i'm still unclear here, is 0,0,0  "facing me" mean the device is lying flat on a table with screen to the sky? or standing on it's end with the screen facing a wall?